File: backend/app/schemas/ai_quality.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel

from app.models.ai_quality import AIQualityEvalType


class AIQualityEvalOut(BaseModel):
    id: uuid.UUID
    session_id: uuid.UUID
    eval_type: AIQualityEvalType
    scores: dict[str, Any]
    summary: str
    created_at: datetime

    model_config = {"from_attributes": True}


class AIQualityEvalsOut(BaseModel):
    transcript_analysis: AIQualityEvalOut | None = None

File: frontend/src/lib/aiQuality.ts
import { api } from "./api";

export type AIQualityEvalType = "transcript_analysis";

export interface AIQualityEval {
  id: string;
  session_id: string;
  eval_type: AIQualityEvalType;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  scores: Record<string, any>;
  summary: string;
  created_at: string;
}

export interface AIQualityEvals {
  transcript_analysis: AIQualityEval | null;
}

export const aiQualityApi = {
  get: (sessionId: string) => api.get<AIQualityEvals>(`/sessions/${sessionId}/ai-quality-evals`),
  run: (sessionId: string) => api.post<AIQualityEvals>(`/sessions/${sessionId}/ai-quality-evals`, {}),
};
